core/store/src/db/rocksdb.rs

```diff
@@ -415,24 +407,74 @@ impl Database for RocksDB {
         }
     }
 
-    fn create_checkpoint(&self, path: &std::path::Path) -> anyhow::Result<()> {
+    fn create_checkpoint(
+        &self,
+        path: &std::path::Path,
+        columns_to_keep: Option<&[DBCol]>,
+    ) -> anyhow::Result<()> {
         let _span =
             tracing::info_span!(target: "state_snapshot", "create_checkpoint", ?path).entered();
         let cp = ::rocksdb::checkpoint::Checkpoint::new(&self.db)?;
-        cp.create_checkpoint(path)?;
+        cp.create_checkpoint(path)
+            .with_context(|| format!("failed to create checkpoint at {}", path.display()))?;
+
+        if let Some(columns_to_keep) = columns_to_keep {
```

Comment:
nit: I would either move it to a helper method or use the `let Some(x) = x else { return ..; };` syntax to return early.